lib/homework3.ex

```diff
@@ -1,18 +1,179 @@
 defmodule Homework3 do
-  @moduledoc """
-  Documentation for `Homework3`.
-  """
+  use Supervisor
 
-  @doc """
-  Hello world.
+  def start() do
+    Supervisor.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def init(_) do
+    Supervisor.init(
+      [
+        {Morning, []}
+      ],
+      strategy: :one_for_one
+    )
+  end
+end
+
+defmodule Morning do
+  use Supervisor
+
+  def start_link(how) do
+    Supervisor.start_link(__MODULE__, how, name: __MODULE__)
+  end
+
+  def init(how) do
+    IO.inspect(how, label: "Woke up")
+    mode = get_mode(how)
+
+    processes = [
+      {Hygiene, mode},
+      {Breakfast, mode},
+      {Clothing, []}
+    ]
+
+    Supervisor.init(processes,
+      strategy: :one_for_one
+    )
+  end
+
+  defp get_mode(:early), do: :slow
+  defp get_mode(:late), do: :fast
+  defp get_mode(_), do: :normal
+end
+
+defmodule Hygiene do
+  @default_counter 50
+  @duration 6000
+  @fee 5
+  use Task
+
+  defmodule Exception do
+    defexception message: "Imaginary parent: Not so fast, my child, #{@fee} more swipes."
+  end
+
+  def start_link(:slow) do
+    Task.start_link(fn -> brush_teeth(60) end)
+  end
+
+  def start_link(:fast) do
+    Task.start_link(fn -> brush_teeth(40) end)
+  end
+
+  def start_link(_) do
+    Task.start_link(fn -> brush_teeth(@default_counter) end)
+  end
+
+  defp clean_a_teeth() do
```

Comment:
If you have at least several teeth, it should be named `clean_teeth`, or `clean_a_tooth` otherwise :))